nixos/modules/services/security/fail2ban.nix

```
@@ -194,60 +218,75 @@ in
         '';
       };
 
-      daemonConfig = mkOption {
-        default = ''
-          [Definition]
-          logtarget = SYSLOG
-          socket    = /run/fail2ban/fail2ban.sock
-          pidfile   = /run/fail2ban/fail2ban.pid
-          dbfile    = /var/lib/fail2ban/fail2ban.sqlite3
+      daemonSettings = mkOption {
+        inherit (configFormat) type;
+
+        defaultText = literalExpression ''
+          '''
+            Definition = {
+              logtarget = "SYSLOG";
+              socket = "/run/fail2ban/fail2ban.sock";
+              pidfile = "/run/fail2ban/fail2ban.pid";
+              dbfile = "/var/lib/fail2ban/fail2ban.sqlite3";
+            };
+          '''
         '';
```

Oops, I went a bit too fast here:

```suggestion
        defaultText = literalExpression ''
          Definition = {
            logtarget = "SYSLOG";
            socket = "/run/fail2ban/fail2ban.sock";
            pidfile = "/run/fail2ban/fail2ban.pid";
            dbfile = "/var/lib/fail2ban/fail2ban.sqlite3";
          };
        '';
```

Note that this won't show up in the manual until https://github.com/NixOS/nixpkgs/pull/230353 is merged.